system/scripts/zone/core/calc_combat.cs

```diff
@@ -197,6 +197,17 @@ public float SCR_CalculateDamage(ICombatEntity attacker, ICombatEntity target, S
 
 		SCR_Combat_AfterCalc(attacker, target, skill, modifier, skillHitResult);
 
+		// TODO: There may be a better place to check this
```

Comment:
The better place is the buff handlers, from where you can hook into all this without having to add the code here^^ Notice the `SCR_Combat_AfterCalc` call just above, which makes the server call the handlers of all active buffs to see if they'd like to change anything about this skill hit.